keras_contrib/layers/advanced_activations.py

```diff
@@ -283,3 +283,75 @@ def get_config(self):
         return dict(list(base_config.items()) + list(config.items()))
 
 get_custom_objects().update({'Swish': Swish})
+
+
+class ReLUs(Layer):
+    """Rectified Linear Unit with Sigmoid to generate oscilations.
+
+    It allows an oscilation in the gradients when the weights are negative.
+    The oscilation can be controlled with a parameter, which makes it be close
+    or equal to zero. So, not all neurons are deactivated and it allows differentiability
+    in more parts of the function.
+
+    # Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model.
+
+    # Output shape
+        Same shape as the input.
+
+    # Arguments
+        epsilon: float. Hyper-parameter used to control oscilations when weights are negative.
+                 The default value, 0.0055, work better for Deep Neural Networks. When using CNNs,
+                 try something around 0.0025.
+
+    # References:
+        - ReLUs: An Alternative to the ReLU Activation Function. This function was
+        first introduced at the Codemotion Amsterdam 2018 and then at the DevDays, in Vilnius, Lithuania.
+        It has been extensively tested with Deep Nets, CNNs, LSTMs, Residual Nets and GANs, based
+        on the MNIST, Kaggle Toxicity and IMDB datasets.
+
+    # Examples
+        The Advanced Activation function ReLUs have to be imported from the
+        keras_contrib.layers package.
+
+        To see full source-code of this architecture and other examples,
+        please follow this link: https://github.com/ekholabs/DLinK
+
+        ```python
+            model = Sequential()
+            model.add(Dense(128, input_shape = (784,)))
+            model.add(ReLUs(epsilon=0.0055))
+            model.add(Dropout(0.2))
+
+            model.add(Dense(256))
+            model.add(ReLUs(epsilon=0.0055))
+            model.add(Dropout(0.3))
+
+            model.add(Dense(1024))
+            model.add(ReLUs(epsilon=0.0055))
+            model.add(Dropout(0.5))
+
+            model.add(Dense(10, activation = 'softmax'))
+        ```
+    """
+
+    def __init__(self, epsilon=0.0055, **kwargs):
+        super(ReLUs, self).__init__(**kwargs)
+        self.supports_masking = True
+        self.epsilon = K.cast_to_floatx(epsilon)
+
+    def call(self, Z):
+        pi = K.variable((np.pi))
```

Comment:
This should be precomputed and stored in an instance variable. Also, since it is a constant, there is no need to assign it to a tensor variable. 

It would be something like self.scale = np.exp(np.sqrt(np.pi)) inside build() and multiply inside call.

Reply:
Hi @titu1994,

Just started looking at the code changes before I got trining everything again.

I agree with the changes, since in this way some of the computations would be done in the __init__/build method and thus avoiding repetition (making things faster).

However, concerning "pi", it need to be a tensor in order to be used by K.exp(K.sqrt(...)). If that's not a tensor, but just a primitive type, we get the following:

```
'float' object has no attribute 'dtype'
```

I'm applying the changes and running tests to make sure all works fine before I push again.

Reply:
Yes, that's why I suggested doing all constant operations in Numpy using no.exp(np.sqrt(np.pi)) * epsilon. 

Multiplying a tensor with a numpy constant is alright. Operating tensor ops on numpy values is not.

Reply:
Ah, got it. I did not pay attention to the syntax and missed the "np" there. :)

Thanks!